### DISCUSSION TITLE
How to change window size and maximize it from init.lua?

### DISCUSSION THREAD

**Discussion by winterwolf** [orbitalquark/textadept#55] [Q&A] [answered]
I am not satisfied with the default window size and changing the **ui.size** values does nothing.
In theory, I could solve this issue by calling wmctrl through os.execute but I don't like such a crutch.
There should be some native solution...

**Comment by orbitalquark** [comment, marked as answer]
Textadept stores window geometry and maximized state in a session file, which is loaded after your *~/.textadept/init.lua* is read, so that's probably what's undoing your settings. Try wrapping your settings in an `events.INITIALIZED` handler:

    events.connect(events.INITIALIZED, function()
      ui.maximized = true
    end)

**Reply by dr-dba** [reply]
(i hope i got the question right, i will try to advice)
it is possible to put in (launch from) init.lua any external utility able to handle the window properties.
In windows just one of multiple choice it is widely known nircmd.
additionally, the window sizing can be set to "full window size" (as the size of desktop) in the launch shortcut.

**Comment by orbitalquark** [comment]
Sorry, I don't understand the question. From your *~/.textadept/init.lua* you can launch processes via `os.spawn()`, `os.execute()`, and `io.popen()`. If you need a window handle or something to pass to such a process, Textadept does not provide that.

**Reply by dr-dba** [reply]
> Sorry, I don't understand the question. 
> From your *~/.textadept/init.lua* you can launch processes via `os.spawn()`, `os.execute()`, and `io.popen()`. 
> If you need a window handle or something to pass to such a process, 
> Textadept does not provide that.

that wasnt a question, that is my attempt to help with answer, my 2 cents :)

yes, from Lua usually you may launch external process with other PId, but.

This external process, if programmable, can say which "parent" process launched it.
It is rather trivial in Windows at least, i guess in *nixes the same or similar.

Further, TA can pass as parameter it's own PID, 
so the utility will accept it and thus knows who is the parent launched by.

Beside that, launched with (for example) "os.execute()",
this new process can just detect TA process by it header title and most recent launch time,
and figure from it "this is my parent which is i am launched by"

Additionally, probably you may build C++ module, 
and use it for detecting own PID, without external tools.

another way, i
magining we use LuaJit (isnt possible in TA), 
or (Lua)CFFI (yess possible in TA), 
thus you may do it without all of it, only with OS API.
(BTW, i can provide the code working for windows)

P.S.
In Windows OS, given the PId (process ID), 
you can find graphic windows (window handlers) opened and belonging to this PID, and vice versa.
Given the window handler, you can handle the window for your pleasure, maximize, minimize, change size, etc ..
